src/utils/segmentInitializer.ts

```diff
@@ -0,0 +1,47 @@
+import { Logger } from './logger';
+import Analytics from 'analytics-node';
+
+export namespace SegmentInitializer {
+  export function initialize(
+    clientSegmentKey: string | undefined = undefined
+  ): Analytics | undefined {
+    const segmentWriteKey: string | undefined =
+      clientSegmentKey || getSegmentWriteKey();
+
+    if (segmentWriteKey) {
+      /* 
+        flushAt: Number ->  The number of messages to enqueue before flushing.
+        flushInterval: Number ->    The number of milliseconds to wait before flushing the queue automatically.
+        ref: https://segment.com/docs/connections/sources/catalog/libraries/server/node/#configuration
+        */
+      let analytics: Analytics = new Analytics(segmentWriteKey, {
+        flushAt: 1,
+        flushInterval: 10000,
+      });
```

Comment:
Since the no of message to be enqueued before flush is set to 1, there is no value in having `flushInterval`.

Reply:
ya, current maybe these values can be discussed and set or could be dropped if not required.